src/systems/userinput/devices/app-aware-touchscreen.js

```diff
@@ -41,10 +41,11 @@ function shouldMoveCursor(touch, raycaster) {
     return true;
   }
   const rawIntersections = [];
+  const canvas = document.querySelector(".a-canvas").getBoundingClientRect();
```

Comment:
This only gets called in touchStart anyway so its not a huge deal, but querySelectors can be really slow so we try to avoid them when we can. Also the use of `left` and `top` on the next lines threw me because of the variable name, didn't notice the `getBoundingClientRect()` at first
```suggestion
  const canvasRect = AFRAME.scense[0].getBoundingClientRect();
```

Reply:
I wanted to use `AFRAME.scene[0]` but for some reason it includes the toolbar as well, whereas the canvas doesn't.
![image](https://user-images.githubusercontent.com/57520910/125160965-37b57800-e1b2-11eb-9fe8-cd8119780ed6.png)
But we can store a ref to canvas to avoid querying it every time.

Reply:
I pushed some changes:
We actually have a ref to canvas in `this.canvas`, and I store canvas rect in `this.canvasRect` and update it on resize using `ResizeObserver`.
Since `shouldMoveCursor` is outside of the object and does not see `this.canvasRect`, I pass rect to the function as well.